src/reducers/pokemonReducer.ts

```diff
@@ -1,82 +1,87 @@
-import { GET_DATA_DONE, GET_DATA_REQUESTED, GET_DATA_FAILED, GET_POKEMON, ADD_TO_FAVORITES, DELETE_FAV_POKEMON } from '../constants';
+import { GET_DATA_DONE, GET_DATA_REQUESTED, GET_DATA_FAILED, GET_POKEMON, DELETE_FAV_POKEMON, ADD_TO_FAVORITES } from '../constants';
 import { handleActions } from 'redux-actions';
 import { Pokemon } from '../model/Pokemon';
+import { combineReducers } from 'redux';
 
-type State = {
-    isLoading: boolean,
-    isError: boolean,
-    pokemonList: Array<Pokemon>,
-    selectedPokemon: Pokemon | null,
-    favoritesPokemons: Array<Pokemon>
-};
+type pokemonList = Array<Pokemon>;
+type isError = boolean;
+type isLoading = boolean;
+type selectedPokemon = Pokemon | null;
+type favoritesPokemons = Array<Pokemon>
 
 const initialState = {
+    pokemonList: [],
     isLoading: false,
     isError: false,
-    pokemonList: [],
     selectedPokemon: null,
     favoritesPokemons: []
+}
+
+const updateFavorites = (favorites, id) => {
+    let newFavorites = favorites.filter(elem => {
+        return (
+            elem.id !== id
+        )
+    });
+    return newFavorites;
```

Comment:
a nawet jeszcze krótszego
```
const updateFavorites = (favorites, id) => favorites.filter(elem => elem.id !== id)
```

Reply:
to w sumie bardziej kwestia upodobań programisty ale wg mnie ten krótszy zapis jest bardziej czytelny i na pewno zmniejsza ilość kodu co w większosci przypadków jest plusem